feature-libs/customer-ticketing/components/customer-ticketing-create/customer-ticketing-create-dialog/customer-ticketing-create-dialog.component.ts

```diff
@@ -0,0 +1,106 @@
+import { Component, ElementRef, HostListener, OnInit } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { MAX_INPUT_CHARACTERS_FOR_MESSAGE } from '@spartacus/customer-ticketing/core';
+import { CustomerTicketingConfig } from '@spartacus/customer-ticketing/root';
+import {
+  FilesFormValidators,
+  FocusConfig,
+  FormUtils,
+  ICON_TYPE,
+  LaunchDialogService,
+} from '@spartacus/storefront';
+
+@Component({
+  selector: 'cx-customer-ticketing-dialog',
+  templateUrl: './customer-ticketing-create-dialog.component.html',
+})
+export class CustomerTicketingCreateDialogComponent implements OnInit {
+  iconTypes = ICON_TYPE;
+  form: FormGroup;
+  inputCharactersLimitForCreateMessage: number = this.getInputCharactersLimit;
+
+  focusConfig: FocusConfig = {
+    trap: true,
+    block: true,
+    autofocus: 'button',
+    focusOnEscape: true,
+  };
+
+  get messagesCharacterLeft(): number {
+    return (
+      this.inputCharactersLimitForCreateMessage -
+      (this.form.get('message')?.value?.length || 0)
+    );
+  }
+
+  get allowedTypes(): string[] | undefined {
+    return this.customerTicketingConfig.customerTicketing
+      ?.attachmentRestrictions?.allowedTypes;
+  }
+
+  get getInputCharactersLimit(): number {
+    return (
+      this.customerTicketingConfig.customerTicketing
+        ?.inputCharactersLimitForCreateMessage ||
+      MAX_INPUT_CHARACTERS_FOR_MESSAGE
+    );
+  }
+
+  get maxSize(): number | undefined {
+    return this.customerTicketingConfig.customerTicketing
+      ?.attachmentRestrictions?.maxSize;
+  }
+
+  get maxEntries(): number | undefined {
+    return this.customerTicketingConfig.customerTicketing
+      ?.attachmentRestrictions?.maxEntries;
+  }
```

Comment:
nit: I feel these methods will be repeated in full or partially on all pages using the AddMessage and the Attachment
Don't you believe it will be a better approach to create a separate component or it  is not worthed? 
Since I don't know Angular, I don't know what would be the implications
@Parthlakhani

Reply:
yes, this methods will be used at 3 places excluding the shared messaging component.
It would be good to create a mechanism that returns these properties altogether from config.
Maybe, it can be done in the component service. `customer-ticketing/components/customer-ticketing.service.ts`

Reply:
What we can do here is create a 'general customer ticketing dialog' that is abstract and that other components will extend to have these functionalities

Reply:
correction to my comment:
It will be used at 2 places excluding the shared messaging component

Reply:
An example would be `BaseMessageComponent`  ---> it's not related to chat component at all if ever :d